build-system/tasks/visual-diff.rb

```diff
@@ -230,17 +329,26 @@ def main()
     createEmptyBuild()
     exit
   end
-  setDebuggingLevel()
-  pid = launchWebServer()
-  if not waitForWebServer()
-    puts red("ERROR: ") + "Failed to start webserver"
+  unless ENV['PERCY_PROJECT'] && ENV['PERCY_TOKEN']
+    puts red("ERROR: ") + "Could not find " + cyan("PERCY_PROJECT") + " and " +
+        cyan("PERCY_TOKEN") + " environment variables."
+    raise 'Missing environment variables'
+  end
+  begin
+    setDebuggingLevel()
+    pid = launchWebServer()
+    if not waitForWebServer()
+      puts red("ERROR: ") + "Failed to start webserver"
+      raise 'Webserver launch failure'
+    end
+    visualTestsConfigJson = loadVisualTestsConfigJson()
+    visualTestsConfig = JSON.parse(visualTestsConfigJson)
+    buildId = runVisualTests(visualTestsConfig)
+    waitForBuildCompletion(buildId)
+    verifyBuildStatus(buildId)
```

Comment:
Would it be possible to split this into two build steps?

1. Start visual diff as early as possible in build
2. At the end run only the polling after other build steps completed, so that the polling time is minimized.

Reply:
Done. I've optimized the number of build steps between creating and verifying the visual diffs, so that we still fail fast if there are diffs, and yet, don't have to wait around for any amount of time.